### PR TITLE
[INJIMOB-1076]: pixelpass swift

### PR DESCRIPTION
**Pull request by abhip2565**

### CODE REVIEW

File: swift/example-qr-generator/Qr/QrApp.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+import pixelpass
+struct QRCodeView: View {
+@State private var inputString = "Test QR Code Generation"
```
Comment (sree96) [inline swift/example-qr-generator/Qr/QrApp.swift]: Can we make the input from user and generate the QR on the fly

File: swift/example-qr-generator/QrTests/QrTests.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
+//
```
Comment (sree96) [inline swift/example-qr-generator/QrTests/QrTests.swift]: I assume this tests are for sample app. Its sample so no need to have tests. Can we remove it?

File: swift/example-qr-generator/QrUITests/QrUITests.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
+//
```
Comment (sree96) [inline swift/example-qr-generator/QrUITests/QrUITests.swift]: I assume this tests are for sample app. Its sample so no need to have tests. Can we remove it?

File: swift/example-qr-generator/QrUITests/QrUITestsLaunchTests.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
+//
+//  QrUITestsLaunchTests.swift
```
Comment (sree96) [inline swift/example-qr-generator/QrUITests/QrUITestsLaunchTests.swift]: I assume this tests are for sample app. Its sample so no need to have tests. Can we remove it?

File: swift/pixelpass/.swiftpm/xcode/xcshareddata/xcschemes/pixelpass 1.xcscheme [hunk 5]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sree96) [inline swift/pixelpass/.swiftpm/xcode/xcshareddata/xcschemes/pixelpass 1.xcscheme:1]: Why there are multiple schemes.?

File: swift/pixelpass/Sources/pixelpass/Zlib.swift [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+//
+
+import Foundation
+import Compression
+class Zlib{
+    func decompress(_ data: Data) -> Data? {
+        if data.count <= 2 {
+            return .init()
+        }
+        let size = 4 * data.count + 8 * 1024
```
Comment (sree96) [inline swift/pixelpass/Sources/pixelpass/Zlib.swift]: Extract magic numbers to const.

File: swift/pixelpass/Sources/pixelpass/Zlib.swift [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            return Data(bytes: buffer, count: read)
+        } as Data
+        buffer.deallocate()
+        return result
+    }
+    
+    
+    func compress(data: String, algorithm: compression_algorithm = COMPRESSION_ZLIB) -> Data? {
+        var sourceBuffer = Array(data.utf8)
+        let destinationBufferSize = sourceBuffer.count + (sourceBuffer.count / 1000) + 12
```
Comment (sree96) [inline swift/pixelpass/Sources/pixelpass/Zlib.swift]: Extract magic numbers to const.

File: swift/pixelpass/Sources/pixelpass/pixelpass.swift [hunk 8]
```diff
@@ -0,0 +1,74 @@
+import Foundation
+import base45_swift
+import CoreImage
+import Compression
+#if canImport(UIKit)
```
Comment (sree96) [inline swift/pixelpass/Sources/pixelpass/pixelpass.swift:5]: remove commented code

File: swift/pixelpass/Sources/pixelpass/pixelpass.swift [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        let compressedData =  Zlib().compress(data:input,algorithm:COMPRESSION_ZLIB)
+        guard let base45EncodedString = compressedData?.toBase45()
+        else{
+            print("Encoding error")
+            return nil;
+        }
+        return base45EncodedString
+    }
+    
+     public func generateQRCode(from string: String, ecc: ECC, header: String = "") -> Data? {
```
Comment (sree96) [inline swift/pixelpass/Sources/pixelpass/pixelpass.swift]: lets have default ecc to be L

**Comment by sree96** [comment]
Add missing documentation also

File: swift/pixelpass/Sources/pixelpass/pixelpass.swift (hunk 8 above)
Comment (abhip2565) [reply swift/pixelpass/Sources/pixelpass/pixelpass.swift:5]: It is required. It is not comment.

File: swift/example-qr-generator/Qr/QrApp.swift (hunk 1 above)
Comment (abhip2565) [reply swift/example-qr-generator/Qr/QrApp.swift]: Done

File: swift/example-qr-generator/QrTests/QrTests.swift (hunk 2 above)
Comment (abhip2565) [reply swift/example-qr-generator/QrTests/QrTests.swift]: done

File: swift/example-qr-generator/QrUITests/QrUITests.swift (hunk 3 above)
Comment (abhip2565) [reply swift/example-qr-generator/QrUITests/QrUITests.swift]: Done

File: swift/example-qr-generator/QrUITests/QrUITestsLaunchTests.swift (hunk 4 above)
Comment (abhip2565) [reply swift/example-qr-generator/QrUITests/QrUITestsLaunchTests.swift]: Done

**Review by sree96 — APPROVED** [review]

**Review by swatigoel — CHANGES_REQUESTED** [review]
Let's keep api interface consistent with JS

**Review by swatigoel — APPROVED** [review]

**Pull request merged**